Wikipedia/Code/DiffController.swift

```diff
@@ -28,20 +23,23 @@ class DiffController {
     let diffThanker: DiffThanker
     let articleTitle: String
     let siteURL: URL
-    lazy var semanticContentAttribute: UISemanticContentAttribute = {
-        let language = siteURL.wmf_language
-        return MWLanguageInfo.semanticContentAttribute(forWMFLanguage: language)
-    }()
     let type: DiffContainerViewModel.DiffType
+    let transformer: DiffTransformer
     
-    init(siteURL: URL, articleTitle: String, diffFetcher: DiffFetcher = DiffFetcher(), revisionFetcher: WMFArticleRevisionFetcher = WMFArticleRevisionFetcher(), globalUserInfoFetcher: GlobalUserInfoFetcher = GlobalUserInfoFetcher(), diffThanker: DiffThanker = DiffThanker(), type: DiffContainerViewModel.DiffType) {
+    init(siteURL: URL, articleTitle: String, diffFetcher: DiffFetcher = DiffFetcher(), revisionFetcher: WMFArticleRevisionFetcher = WMFArticleRevisionFetcher(), globalUserInfoFetcher: GlobalUserInfoFetcher = GlobalUserInfoFetcher(), diffThanker: DiffThanker = DiffThanker(), type: DiffContainerViewModel.DiffType, transformer: DiffTransformer? = nil) {
         self.diffFetcher = diffFetcher
         self.revisionFetcher = revisionFetcher
         self.globalUserInfoFetcher = globalUserInfoFetcher
         self.diffThanker = diffThanker
         self.articleTitle = articleTitle
         self.siteURL = siteURL
         self.type = type
+        
+        if let transformer = transformer {
+            self.transformer = transformer
```

Comment:
it looks like `transformer` is never passed as an argument - is it necessary to account for that case or could it be always created here?

Reply:
@nambatee not necessary as it is now, was just trying to make it injectable for future unit testing with a mock transformer. I can remove until I have actual unit tests in place, pretty sure most of these injections aren't needed yet.